app/models/katello/repository.rb

```diff
@@ -207,6 +206,16 @@ def self.repo_path_from_content_path(environment, content_path)
       "#{path_prefix}/#{path}"
     end
 
+    def content_id
+      # Currently deb content will store a content_id on each Reposiotry, while all other content
+      # types will store one on the RootRepository.
+      self[:content_id] || root.content_id
+    end
```

Comment:
The basic idea here is that all content types will now look for the content_id on `Katello::Repository` instead of `Katello::RootRepository`. However, for all non-deb content types, the content_id on a `Katello::Repository` will always be `nil`, and will default right back to `root.content_id`. That way, things don't actually change much for non-deb content types. They will continue to store and use a single `content_id` per root repository.